flang/include/flang/Optimizer/CodeGen/CGOps.td

```diff
@@ -15,7 +15,7 @@
 #define FORTRAN_DIALECT_FIRCG_OPS
 
 include "mlir/IR/SymbolInterfaces.td"
-include "flang/Optimizer/TypePredicates.td"
+include "flang/Optimizer/Dialect/FIRTypes.td"
```

Comment:
The reason TypePredicates is located in Optimizer and not Optimizer/Dialect is that they are shared with the codegen dialect.
In fact, how does this compile since TypePredicates.td is still used there?

Reply:
`TypePredicates.td` information is now merged in `flang/Optimizer/Dialect/FIRTypes.td`. Since `FIROps.td` does not have types information anymore the the new `FIRTypes.td` can be shared between the two dialects.